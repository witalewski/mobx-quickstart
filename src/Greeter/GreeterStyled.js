import styled from '@emotion/styled';

export const GreeterStyled = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
