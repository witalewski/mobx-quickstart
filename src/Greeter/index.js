export { default as Greeter } from './Greeter';
