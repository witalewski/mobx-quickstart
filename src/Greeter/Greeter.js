import React, { Component, createRef } from 'react';
import { observer, inject } from 'mobx-react';
import { string, func } from 'prop-types';
import { GreeterStyled } from './GreeterStyled';

class Greeter extends Component {
  static propTypes = {
    greeting: string.isRequired,
    name: string.isRequired,
    setName: func.isRequired,
  };

  constructor() {
    super();
    this.input = createRef();
  }

  componentDidMount() {
    this.input.current.focus();
  }

  onNameChange = ({ target: { value } }) => {
    this.props.setName(value);
  };

  render() {
    return (
      <GreeterStyled>
        <h1>{this.props.greeting}</h1>
        <div>
          Your name:{' '}
          <input
            ref={this.input}
            type="text"
            value={this.props.name}
            onChange={this.onNameChange}
          />
        </div>
      </GreeterStyled>
    );
  }
}

export { Greeter };
export default inject(({ appState }) => ({
  greeting: appState.greeting,
  name: appState.name,
  setName: appState.setName,
}))(observer(Greeter));
