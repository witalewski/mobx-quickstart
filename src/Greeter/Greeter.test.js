import React from 'react';
import { Greeter } from './Greeter';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Greeter', () => {
  it('displays greeting', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Greeter name="Kris" greeting="Hello, Kris" setName={jest.fn()} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
