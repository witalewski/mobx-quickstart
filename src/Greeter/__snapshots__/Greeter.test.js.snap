// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter displays greeting 1`] = `
<ForwardRef(render)>
  <h1>
    Hello, Kris
  </h1>
  <div>
    Your name:
     
    <input
      onChange={[Function]}
      type="text"
      value="Kris"
    />
  </div>
</ForwardRef(render)>
`;
